### ISSUE TITLE
ServiceMonitor for prometheus does not seem to be installed

### ISSUE THREAD

**Issue by ilya-scale** [kedacore/charts#391]
I have this configuration for values for the helm chart

```yaml
    image:
      keda:
        tag: 2.9.3 # TODO there is a critical bug in 2.9.2 which ships with 2.9.3 chart. When a new chart comes in we can update and delete image/keda/tag
    
    podIdentity:
      azureWorkload:
        enabled: true
        clientId: SomeId
    
    prometheus:
      operator:
        enabled: true
        serviceMonitor:
          enabled: true
          namespace: prometheus # By default, Prometheus discovers PodMonitors and ServiceMonitors within its namespace, that are labeled with the same release tag as the prometheus-operator release.
          additionalLabels:
            release: kube-prometheus-stack # See comment above
```

It seems that operator service does get necessary /metrics, but the ServiceMonitor is not created for some reason. I have cloned the repository for the latest version and tried to run helm template myself, then the required manifest was rendered just fine.

P.S. Can it be that these are the newest changes that made it to the docs but not to the actual release?

## Expected Behavior

ServiceMonitor resource is generated

## Actual Behavior

No ServiceMonitor resource is generated

## Steps to Reproduce the Problem

  1. Use the provided values
  2. Install helm chart
  3. Observe no ServiceMonitor

## Specifications

  - **KEDA Version:** 2.9.3 (both chart and keda)
  - **Kubernetes Version:** 1.24

P.S. I have updated to chart 2.9.4 and the same problem is still there unfortunately.

**Comment by ChauAnhTuan** [comment]
I have the same issue.

**Comment by JorTurFer** [comment]
Hi,
Sadly, you can't check `main` branch if you are using v2.9. `main` branch contains the changes that will be released in next KEDA release. For example, if you use main, you will see that webhooks are also deployed, but they aren't released yet.

In this case, I'd suggest checking the branch `release/v2.9` because all the commits in that branch are available in latest v2.9 version. The tag v2.9.4 was generated from `main` by error instead of `release/v2.9` (it's already solved), but as I said, for latest features in an specific version, I suggest checking the release branch directly, `main` isn't a good reference for it since you are not using KEDA from `main` either.

You can expect serviceMonitor support as part of KEDA v2.10 in some weeks. In the meantime, I'm goign to close this issue as not planned. If you think that this is an issue that needs to be solved as part of v2.9, feel free to reopen it and we can discuss it
